src/file/Lock.php

```diff
@@ -36,7 +36,7 @@ public function __construct(
     /* HH_IGNORE_ERROR[4107] __PHPStdLib */
     $flock_result = \flock($this->resource, $lock_type, inout $_wouldblock);
     if (!$flock_result) {
-      throw new LockAcquisitionException(OS\_Private\errno() as nonnull);
+      OS\_Private\throw_errno(OS\_Private\errno() as nonnull, "flock failed");
```

Comment:
Maybe one of these would be preferable:

- `throw_errno()` does the `as nonnull` cast inside
- a separate function like `throw_errno_x()` which does that
- maybe we should have an IO/OS exception subclass to handle the case of a null errno? It's unlikely but if it somehow happens, at least we'd still have a relevant error message ("fopen failed") instead of "expected nonnull" or whatever `as` throws